app/src/App.vue

```diff
@@ -0,0 +1,199 @@
+                            </el-form-item>
+
+                            <el-form-item :label="translations.action" prop="selectedAction">
+                                <el-select v-model="form.selectedAction" :placeholder="translations.action" clearable>
+                                    <el-option label="Go to cart" value="1"></el-option>
+                                    <el-option label="Check out" value="2"></el-option>
+                                </el-select>
+                            </el-form-item>
+
+                            <el-form-item :label="translations.sharableLink" class="form-link">
+                                <span v-if="form.selectedProduct && form.selectedAction" class="generated-link">{{ ps_base_url }}/index.php?fc=module&module=ps_buybuttonlite&controller=RedirectManager&id_product={{ form.selectedProduct.id_product }}&action={{ form.selectedAction }}<label v-if="form.selectedProduct.id_product_attribute">&id_product_attribute={{ form.selectedProduct.id_product_attribute }}</label></span>
+                                <span v-else class="no-link">Please select a product and an action</span>
+                            </el-form-item>
+                        </el-form>
+                    </el-col>
+                </el-row>
+            </PsCardBlock>
+
+            <PsCardFooter>
+                <template slot="footerRight">
+                    <el-button type="primary" @click="validateForm('form')">{{ translations.copyToClipboard }}</el-button>
+                </template>
+            </PsCardFooter>
+        </PsCard>
+
+        <BannerPromo
+            :id_product_addons="26835"
+            :iso_code="context.country.iso_code"
+            :iso_lang="context.language.iso_code"
+            :ps_version="ps_version"
+            :trackingAddonsLink="trackingAddonsLink"
+        >
+            <el-row>
+                <el-col :span="12">
+                    <span class="features"><i class="material-icons">add_circle</i> Card & Button design integration</span>
+                </el-col>
+                <el-col :span="12">
+                    <span class="features"><i class="material-icons">add_circle</i> Overview and Card Manager</span>
+                </el-col>
+            </el-row>
+            <el-row>
+                <el-col :span="12">
+                    <span class="features"><i class="material-icons">add_circle</i> Statistics</span>
+                </el-col>
+                <el-col :span="12">
+                    <span class="features"><i class="material-icons">add_circle</i> Virtual cart on external pages</span>
+                </el-col>
+            </el-row>
+        </BannerPromo>
+    </div>
+</template>
+
+<script>
+import axios from 'axios'
+import PsCard from '@/components/PsCard.vue'
+import PsCardHeader from '@/components/PsCardHeader.vue'
+import PsCardBlock from '@/components/PsCardBlock.vue'
+import PsCardFooter from '@/components/PsCardFooter.vue'
```

Comment:
Is this application going to grow ? If yes (and in 99% of the time), it is a good way to think in a 'component way'.
Here, you create a component for each part of your layout application. This is a good habit to have.
But in the case of this application, since:
 1- it's a very small app
 2- the fact that I don't think it is going to grow in a bigger application 
 3- and the fact that all of Ps* components have a very basic behavior
it would easier to implement the behavior of those components directly in this file, without creating a file every time. It would be as easy to understand as the current behavior, and will make your code lighter. Also, less files, less imports, better optimization.

And if the app would grow, then, it would be better to have a parent state (with vue-router), to handle all the common layout.